lib/services/downloads_service_backend.dart

```diff
@@ -0,0 +1,1515 @@
+            _isar.writeTxnSync(() {
+              _downloadsService.updateItemState(task, DownloadItemState.failed);
+            });
+            continue;
+          }
+          // Do not limit enqueued downloads on IOS, it throttles them like crazy on its own.
+          while (_activeDownloads.length >=
+                      FinampSettingsHelper
+                          .finampSettings.maxConcurrentDownloads &&
+                  !Platform.isIOS ||
+              _finampUserHelper.currentUser == null) {
+            await Future.delayed(const Duration(milliseconds: 500));
+          }
+          await SchedulerBinding.instance.scheduleTask(() {
+            _activeDownloads.add(task.isarId);
+            // Base URL shouldn't be null at this point (user has to be logged in
+            // to get to the point where they can add downloads).
+            var url = switch (task.type) {
+              DownloadItemType.song => _jellyfinApiData
+                  .getSongDownloadUrl(
+                      item: task.baseItem!,
+                      transcodingProfile: task.fileTranscodingProfile)
+                  .toString(),
+              DownloadItemType.image => _jellyfinApiData
+                  .getImageUrl(
+                    item: task.baseItem!,
+                    // Download original file
+                    quality: null,
+                    format: null,
+                  )
+                  .toString(),
+              _ => throw StateError("???"),
+            };
+            _enqueueLog.fine(
+                "Submitting download ${task.name} to background_downloader.");
+            var downloadTask = DownloadTask(
+                taskId: task.isarId.toString(),
+                url: url,
+                //requiresWiFi:
+                //    FinampSettingsHelper.finampSettings.requireWifiForDownloads,
+                displayName: task.name,
+                baseDirectory:
+                    task.fileDownloadLocation!.baseDirectory.baseDirectory,
+                retries: 3,
+                directory: path_helper.dirname(task.path!),
+                headers: {
+                  "Authorization": _finampUserHelper.authorizationHeader,
+                },
+                filename: path_helper.basename(task.path!));
+            return Future.sync(() async {
+              //bool success = await FileDownloader().resume(downloadTask);
+              //if (!success) {
+              bool success = await FileDownloader().enqueue(downloadTask);
+              //}
+              if (!success) {
+                // We currently have no way to recover here.  The user must re-sync to clear
+                // the stuck download.
+                _enqueueLog.severe(
+                    "Task ${task.name} failed to enqueue with background_downloader.");
+              }
+            });
+            // Set priority high to prevent stalling
+          }, Priority.animation + 50);
```

Comment:
If the priority is higher than `Priority.animation`, won't this cause animations to lag? I'd much prefer a tad worse performance but a fluent UI...

Reply:
I don't think any built in elements of flutter actually use this scheduler, at least as far as I can tell.  I think it would only affect animations we create driven by this function, and we don't have any.  Any task set below animation has a chance of being delayed almost indefinitely with the default scheduler, so I just set everything above that.

Reply:
Interesting, that's crappy :(